lib/SILOptimizer/Transforms/Outliner.cpp

```diff
@@ -0,0 +1,1306 @@
+      Apply->getNumArguments() != 2 || Apply->getArgument(0) != Enum ||
+      Apply->getArgument(1) != Metatype ||
+      Apply->getSubstCalleeType()->getNumResults() != 1)
+    return false;
+  if (Apply->getSubstCalleeType()->getSingleResult().getConvention() !=
+      ResultConvention::Owned)
+    return false;
+
+  // Check that we call the _unconditionallyBridgeFromObjectiveC witness.
+  auto NativeType = Apply->getType().getSwiftRValueType();
+  auto *BridgeFun = FunRef->getReferencedFunction();
+  auto *SwiftModule = BridgeFun->getModule().getSwiftModule();
+  auto bridgeWitness = getBridgeFromObjectiveC(NativeType, SwiftModule);
+  if (BridgeFun->getName() != bridgeWitness.mangle())
+    return false;
+
+  // %41 = enum $Optional<String>, #Optional.some!enumelt.1, %40 : $String
+  ADVANCE_ITERATOR_OR_RETURN_FALSE(It);
+  Enum = dyn_cast<EnumInst>(It);
+  if (!Enum || !Enum->hasOneUse() || !Enum->hasOperand() ||
+      Enum->getOperand() != Apply)
+    return false;
+
+  // br bb10(%41 : $Optional<String>)
+  ADVANCE_ITERATOR_OR_RETURN_FALSE(It);
+  Br = dyn_cast<BranchInst>(It);
+  if (!Br || Br->getDestBB() != MergeBB || Br->getNumArgs() != 1 ||
+      Br->getArg(0) != Enum)
+    return false;
+  return true;
+}
+
+bool BridgedProperty::matchInstSequence(SILBasicBlock::iterator It) {
+  // Matches:
+  // [ optionally:
+  //    %31 = load %30 : $*UITextField
+  //    strong_retain %31 : $UITextField
+  // ]
+  //    %33 = class_method [volatile] %31 : $UITextField, #UITextField.text!getter.1.foreign : (UITextField) -> () -> String?, $@convention(objc_method) (UITextField) -> @autoreleased Optional<NSString>
+  //    %34 = apply %33(%31) : $@convention(objc_method) (UITextField) -> @autoreleased Optional<NSString>
+  //    switch_enum %34 : $Optional<NSString>, case #Optional.some!enumelt.1: bb8, case #Optional.none!enumelt: bb9
+  //
+  //  bb8(%36 : $NSString):
+  //    %37 = function_ref @_T0SS10FoundationE36_unconditionallyBridgeFromObjectiveCSSSo8NSStringCSgFZ : $@convention(method) (@owned Optional<NSString>, @thin String.Type) -> @owned String
+  //    %38 = enum $Optional<NSString>, #Optional.some!enumelt.1, %36 : $NSString
+  //    %39 = metatype $@thin String.Type
+  //    %40 = apply %37(%38, %39) : $@convention(method) (@owned Optional<NSString>, @thin String.Type) -> @owned String
+  //    %41 = enum $Optional<String>, #Optional.some!enumelt.1, %40 : $String
+  //    br bb10(%41 : $Optional<String>)
+  //
+  //  bb9:
+  //    %43 = enum $Optional<String>, #Optional.none!enumelt
+  //    br bb10(%43 : $Optional<String>)
+  //
+  //  bb10(%45 : $Optional<String>):
+
+  clearState();
+
+  // %31 = load %30 : $*UITextField
+  auto *Load = dyn_cast<LoadInst>(It);
+
+  // Otherwise, trying matching from the method call.
+  if (!Load) {
+    // Try to match without the load/strong_retain prefix.
+    auto *CMI = dyn_cast<ClassMethodInst>(It);
+    if (!CMI)
+      return false;
+    FirstInst = CMI;
+  } else FirstInst = Load;
```

Comment:
The way this statement is at the end of the line after the else seems weird... did you intend for this? Or did git-clang-format do this?

Reply:
No clang-format puts it on the next line.